uint128.go

```diff
@@ -452,3 +450,43 @@ func FromString(s string) (u Uint128, err error) {
 	_, err = fmt.Sscan(s, &u)
 	return
 }
+
+// MarshalText implements encoding.TextMarshaler.
+func (u Uint128) MarshalText() ([]byte, error) {
+	return []byte(u.String()), nil
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler.
+func (u *Uint128) UnmarshalText(text []byte) error {
+	str := processString(string(text))
+	v, err := FromString(str)
+	if err != nil {
+		return err
+	}
+	*u = v
+	return nil
+}
+
+// MarshalJSON implements json.Marshaler.
+func (u Uint128) MarshalJSON() ([]byte, error) {
+	return []byte(u.String()), nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (u *Uint128) UnmarshalJSON(data []byte) error {
+	return u.UnmarshalText(data)
+}
```

Comment:
I don't think it's necessary to implement these if `MarshalText` and `UnmarshalText` are defined. `json.Marshal` will check for a `MarshalText` method, and likewise `json.Unmarshal` with `UnmarshalText`. (Worth confirming this with a test, though)